### PR TITLE
fix: set min and max protocol length

### PR DESCRIPTION
**Pull request by wemeetagain**
- Prevent unnecessarily large protocols
- Prevent 0-length protocols

### CODE REVIEW

File: src/multistream.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ import type { AbortOptions } from '@libp2p/interfaces'
 import type { Source } from 'it-stream-types'
 import type { Reader } from 'it-reader'
 import type { MultistreamSelectInit } from '.'
+import { MAX_PROTOCOL_LENGTH } from './constants'
```
Comment (achingbrain) [inline src/multistream.ts:15]:
```suggestion
import { MAX_PROTOCOL_LENGTH } from './constants.js'
```

**Review by dapplion — APPROVED** [review]

**Comment by mpetrunic** [comment]
@wemeetagain Seems like tests are failing but not by accident, errors changed to UNDER_READ

**Review by dapplion — APPROVED** [review]

**Comment by achingbrain** [comment]
@wemeetagain could you take a look at the failing tests please then we can get this in

**Comment by wemeetagain** [comment]
thanks @mpetrunic <3

**Review by mpetrunic — APPROVED** [review]

**Pull request merged**